### PR TITLE
CDAP-14592 Cleanup unit-test to be Flow free

### PR DESCRIPTION
**Pull request by chtyim**
- For tests that only testing Flow specific functionalities, they are removed
- Some tests are replaced with either Service or Worker for equivalent functional tests
- Tests that use app containing Flow, but actually don't need flow, they are replaced with AllProgramsApp

### CODE REVIEW

File: cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java [hunk 1]
```diff
@@ -127,12 +125,9 @@
   private static final ApplicationId FAKE_APP_ID_V_1 = NamespaceId.DEFAULT.app(FakeApp.NAME, V1_SNAPSHOT);
   private static final ArtifactId FAKE_PLUGIN_ID = NamespaceId.DEFAULT.artifact(FakePlugin.NAME, V1);
   private static final ProgramId FAKE_WORKFLOW_ID = FAKE_APP_ID.workflow(FakeWorkflow.NAME);
-  private static final ProgramId FAKE_FLOW_ID = FAKE_APP_ID.flow(FakeFlow.NAME);
   private static final ProgramId FAKE_SPARK_ID = FAKE_APP_ID.spark(FakeSpark.NAME);
-  private static final ServiceId PING_SERVICE_ID = FAKE_APP_ID.service(PingService.NAME);
```
Comment (anew) [inline cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java:132]: why remove this one?

File: cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java [hunk 2]
```diff
@@ -1043,12 +1043,6 @@ public ProgramScheduleStatus call() throws Exception {
     }, 5, TimeUnit.SECONDS, 30, TimeUnit.MILLISECONDS);
   }
 
-  @Category(XSlowTests.class)
```
Comment (anew) [inline cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java:1068]: +1 for removing redundant tests.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/ApplicationLifecycleServiceTest.java [hunk 3]
```diff
@@ -70,6 +67,7 @@ public static void setup() throws Exception {
     applicationLifecycleService = getInjector().getInstance(ApplicationLifecycleService.class);
     locationFactory = getInjector().getInstance(LocationFactory.class);
     artifactRepository = getInjector().getInstance(ArtifactRepository.class);
+
```
Comment (anew) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/ApplicationLifecycleServiceTest.java]: extra newline?

File: cdap-client-tests/src/test/java/co/cask/cdap/client/ClientTestsSuite.java [hunk 4]
```diff
@@ -29,22 +29,22 @@
 @Category(XSlowTests.class)
 @RunWith(TestSuite.class)
 @Suite.SuiteClasses({
-  ApplicationClientTestRun.class,
-  ArtifactClientTestRun.class,
-  DatasetClientTestRun.class,
-  LineageHttpHandlerTestRun.class,
-  MetaClientTestRun.class,
+//  ApplicationClientTestRun.class,
```
Comment (anew) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/ClientTestsSuite.java]: why comment out and not remove?

File: cdap-client-tests/src/main/java/co/cask/cdap/client/app/ConfigTestApp.java [hunk 5]
```diff
@@ -1,153 +0,0 @@
...
-import com.google.common.base.Preconditions;
-import com.google.common.base.Strings;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import java.io.IOException;
-import java.util.concurrent.TimeUnit;
-
-/**
- * Test Application to test passing in Configuration during Application creation time.
```
Comment (anew) [inline cdap-client-tests/src/main/java/co/cask/cdap/client/app/ConfigTestApp.java:39]: +1 for removing duplication!

**Review by anew — APPROVED** [review]
Only a few comments.
Also, please update copyright to 2019 for all changed files.

File: cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java (hunk 1 above)
Comment (chtyim) [reply cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java:132]: It is an unused variable as warned by the IDE.

File: cdap-client-tests/src/test/java/co/cask/cdap/client/ClientTestsSuite.java (hunk 4 above)
Comment (chtyim) [reply cdap-client-tests/src/test/java/co/cask/cdap/client/ClientTestsSuite.java]: Oh, unintended changes for running tests locally.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/ApplicationLifecycleServiceTest.java (hunk 3 above)
Comment (chtyim) [reply cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/ApplicationLifecycleServiceTest.java]: fixed

**Comment by chtyim** [comment]
Squashed all changes. Building in https://builds.cask.co/browse/CDAP-DUT6707-9

**Comment by chtyim** [comment]
Build passed

**Pull request merged**